### ISSUE TITLE
[nginx*] [suggestion] put nginx cryptpad.conf outside of the docker-entrypoint.sh

### ISSUE THREAD

**Issue by Loader009** [xwiki-labs/cryptpad-docker#41]
I think that getting everything into docker-entrypoint.sh is too much and error-prone.
I suggest to get the config into a separate file, it would be easier to maintain and it would take less time to test.

It would also ease the reading for anyone who isn't completely aware of the sed-syntax - as well as no need for looking for any example, which then has to be modified and the modifications of sed have to be understood then.

TLDR; right now it would take a considerable amount of time to read/understand how cryptpad.conf is being created.

**Comment by almereyda** [comment]
Yes, the `sed`ing is quite a clutch. But from a container PoV, it is the reasonable way to create the configuration from environment on first container start up.

https://github.com/xwiki-labs/cryptpad-docker/blob/72dd7030c1dc1c70b5ff3f53b8451f5af19a2927/docker-entrypoint.sh#L41-L43

Please note that it still allows you to mount your own configuration file to /etc/nginx/conf.d/cryptpad.conf

Other Compose projects generate configuration files sometimes with leveraging `envsubst` or other templating engines.

Nginx also supports templating syntax to derive configuration from environment, see

- https://github.com/docker-library/docs/blob/86b54b5f18a67802dac17c96310e233f1647630c/nginx/README.md#using-environment-variables-in-nginx-configuration-new-in-119
- https://stackoverflow.com/a/62844707/1959568

I think the case to allow running behind multiple load balancers #36 asks us here to leverage a templating enginge with little logic blocks, like Jinja2, to create the configuration file in this specific case. More generic use cases could profit from native Nginx templating, since this reduces the boilerplate neccessary to get the application running. Less is more in this case.
